File: apps/backend/app/schemas/stylist_runtime_settings.py
from pydantic import BaseModel, Field

from app.domain.stylist_runtime_settings import (
    MAX_DAILY_CHAT_SECONDS_LIMIT_NON_ADMIN,
    MAX_DAILY_GENERATION_LIMIT_NON_ADMIN,
    MAX_MESSAGE_COOLDOWN_SECONDS,
    MAX_TRY_OTHER_STYLE_COOLDOWN_SECONDS,
)
from app.schemas.common import TimestampedRead


class StylistRuntimeSettingsUpdate(BaseModel):
    daily_generation_limit_non_admin: int = Field(ge=0, le=MAX_DAILY_GENERATION_LIMIT_NON_ADMIN)
    daily_chat_seconds_limit_non_admin: int = Field(ge=0, le=MAX_DAILY_CHAT_SECONDS_LIMIT_NON_ADMIN)
    message_cooldown_seconds: int = Field(ge=0, le=MAX_MESSAGE_COOLDOWN_SECONDS)
    try_other_style_cooldown_seconds: int = Field(ge=0, le=MAX_TRY_OTHER_STYLE_COOLDOWN_SECONDS)


class StylistRuntimeSettingsRead(TimestampedRead):
    id: int
    daily_generation_limit_non_admin: int
    daily_chat_seconds_limit_non_admin: int
    message_cooldown_seconds: int
    try_other_style_cooldown_seconds: int

File: apps/frontend/src/widgets/admin/model/stylistRuntimeSettings.ts
import type { StylistRuntimeSettings } from "@/shared/api/types";

export const MAX_DAILY_GENERATION_LIMIT_NON_ADMIN = 1000;
export const MAX_DAILY_CHAT_SECONDS_LIMIT_NON_ADMIN = 86_400;
export const MAX_MESSAGE_COOLDOWN_SECONDS = 3_600;
export const MAX_TRY_OTHER_STYLE_COOLDOWN_SECONDS = 3_600;

export type StylistRuntimeSettingsUpdatePayload = Pick<
  StylistRuntimeSettings,
  | "daily_generation_limit_non_admin"
  | "daily_chat_seconds_limit_non_admin"
  | "message_cooldown_seconds"
  | "try_other_style_cooldown_seconds"
>;

export function toBoundedInteger(value: string | number, fallback: number, min: number, max: number) {
  if (typeof value === "number" && Number.isFinite(value)) {
    return Math.min(Math.max(Math.round(value), min), max);
  }
  return fallback;
}

export function buildStylistRuntimeSettingsUpdatePayload(
  settings: StylistRuntimeSettings
): StylistRuntimeSettingsUpdatePayload {
  return {
    daily_generation_limit_non_admin: settings.daily_generation_limit_non_admin,
    daily_chat_seconds_limit_non_admin: settings.daily_chat_seconds_limit_non_admin,
    message_cooldown_seconds: settings.message_cooldown_seconds,
    try_other_style_cooldown_seconds: settings.try_other_style_cooldown_seconds,
  };
}
